### PR TITLE
[TEST] Add streaming chunk-edge coverage to Gateway integration suite

### PR DESCRIPTION
**Pull request by bolubo**
## Pull Request Description

Adds Gateway integration coverage for the four streaming chunk-edge cases listed in #2708, using the existing `test/integration/gateway` fixture:

- [x] Multiple SSE events in one callback chunk.
- [x] UTF-8 content split across callback chunks.
- [x] Empty end-of-stream chunks, including an empty EOS chunk that flushes a partial line buffered from the previous chunk.
- [x] Usage extraction from a final streaming chunk while preserving the original body.

The specs go through the real ext_proc `Server.Process` boundary and pin the callback acks plus the finalized token counts, so a regression in tail reassembly, usage extraction, or response pass-through shows up as a failing spec.

`fixture.go` now records the token counts passed to `DoneRequestTrace`, which lets the usage cases assert 3/5 and 5/7 instead of header presence only. Existing expectations are unchanged, and the parser matrices remain in the unit tests.

The fixture doesn't run Envoy, so the forwarded bytes are checked at the callback boundary rather than on the wire; the Envoy-side cases stay with the E2E follow-ups.

Verified locally:
- `go test ./test/integration/gateway/... -count=1` (36/36 specs)
- `go test -race ./test/integration/gateway/... -count=1`
- `go vet ./test/integration/gateway/...`
- `gofmt -l test/integration/gateway/` (no output) and `git diff --check`

## Related Issues

Related to #2708. The issue tracks the remaining follow-ups, so it stays open.

### CODE REVIEW

File: test/integration/gateway/streaming_chunk_edge_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	})
+
+	It("reassembles a UTF-8 character split across callback chunks", func() {
+		fixture := newGatewayFixture([]*corev1.Pod{readyPod("target", "10.0.0.2", nil)})
+		payload := "data: {\"id\":\"utf8\",\"choices\":[{\"delta\":{\"content\":\"你好，世界\"}}]," +
+			"\"usage\":{\"prompt_tokens\":3,\"completion_tokens\":5,\"total_tokens\":8}}\n\n" +
+			"data: [DONE]\n\n"
+		// Split inside a multi-byte character so the second chunk starts mid-rune.
+		split := strings.Index(payload, "好") + 1
+		first, second := payload[:split], payload[split:]
```
[the remark this answers is not in the document]
Comment (bolubo) [reply test/integration/gateway/streaming_chunk_edge_test.go:59]: Fixed in 17dab84: the spec now asserts the split rune is present before slicing.

**Comment by bolubo** [comment]
Fixed in `a3e5738`: the empty end-of-stream case now also covers a partial line buffered from the previous chunk being flushed by the empty EOS chunk, asserting the extracted tokens and the untouched callbacks (the UTF-8 tightening landed in `1b29800`).

Re the `inputs[N]` assertions: kept to match `streaming_test.go`'s pattern; the pass-through guarantee comes from `BodyMutation == nil` in `expectPreservedStreamingResponses`.

**Review by varungup90 — APPROVED** [review]
